### PR TITLE
Added InterfaceRedundancyGroupAssociation to GraphQL.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #4796 
# What's Changed

1. Added `InterfaceRedundancyGroupAssociation` to GraphQL.
2. Added try...except... block to handle the `AttributeError` raised by `get_absolute_url()` when a valid absolute url is not found.

- [x] I should open a PR against develop as well for the second item.
- [x] IPAddressToInterface
- [x] VRFDeviceAssignment
- [x] VRFPrefixAssignment

Screen Shot 2023-11-27 at 2 50 29 PM
Screen Shot 2023-11-27 at 6 19 19 PM

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/4796.added [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Added InterfaceRedundancyGroupAssociation to GraphQL.
+Added try...except... block to handle the `AttributeError` raised by `get_absolute_url()` when a valid absolute url is not found.
```
Comment (glennmatthews) [inline changes/4796.added:2]: Not sure I see the connection between these two items. Since as you mentioned the try/except logic may be needed in `develop`, is there a reason it shouldn't just be part of a separate PR that can bubble up to `next` in due time?

File: nautobot/dcim/models/device_components.py [hunk 2]
```diff
@@ -849,7 +849,9 @@ def remove_interface(self, interface):
         )
         return instance.delete()
 
-
+@extras_features(
+    "graphql"
+)
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/dcim/models/device_components.py:854]: This seems correct for what it is (allowing this model to be queried directly in GraphQL). Wonder if it's sufficient though - we probably also need to make sure there's a pattern in place for traversing in GraphQL from an Interface to its InterfaceRedundancyGroupAssociations, from an InterfaceRedundancyGroup to its InterfaceRedundancyGroupAssociations, and vice versa - may require some custom GraphQL types, or might not.

File: nautobot/dcim/views.py [hunk 3]
```diff
@@ -2910,3 +2911,15 @@ class InterfaceRedundancyGroupAssociationUIViewSet(ObjectEditViewMixin, ObjectDe
     form_class = forms.InterfaceRedundancyGroupAssociationForm
     template_name = "dcim/interfaceredundancygroupassociation_create.html"
     lookup_field = "pk"
+
+    def get_return_url(self, request, obj=None):
+        # First, see if `return_url` was specified as a query parameter or form data. Use this URL only if it's
+        # considered safe.
```
Comment (glennmatthews) [inline nautobot/dcim/views.py:2917]: What's this about? What case are we protecting against?

File: nautobot/dcim/models/device_components.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/dcim/models/device_components.py:854]: It seems good as far as traversal goes.
Screen Shot 2023-11-27 at 6 05 11 PM
Screen Shot 2023-11-27 at 6 07 27 PM

File: changes/4796.added (hunk 1 above)
Comment (HanlinMiao) [reply changes/4796.added:2]: Yeah confirmed the error in develop as well, will just open a PR in `develop`

File: nautobot/dcim/views.py (hunk 3 above)
Comment (HanlinMiao) [reply nautobot/dcim/views.py]: That was for debugging purposes, removed now.

**Review by glennmatthews — COMMENTED** [review]
Changes look reasonable, anything we can add in unit tests?

**Review by glennmatthews — APPROVED** [review]
